### PR TITLE
Feat/global rate limited exit

### PR DESCRIPTION
**Pull request by ElliotFriedman**

### CODE REVIEW

File: contracts/core/IPermissionsV2.sol [hunk 1]
```diff
@@ -26,6 +26,10 @@ interface IPermissionsV2 is IAccessControl {
 
     function grantRateLimitedRedeemer(address rateLimitedRedeemer) external;
 
+    function grantRateLimitedDepleter(address rateLimitedRedeemer) external;
+
+    function grantRateLimitedReplenisher(address rateLimitedRedeemer) external;
+
```
Comment (eswak) [inline contracts/core/IPermissionsV2.sol]: Deplete/Replenish are generic terms that apply to any buffer, could we find a better name for these system exits ?

File: contracts/limiter/GlobalSystemExitRateLimiter.sol [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+    /// @param amount the amount of dollars to deplete the buffer by
+    function depleteBuffer(
+        uint256 amount
+    )
+        external
+        /// checks
+        onlyVoltRole(VoltRoles.VOLT_RATE_LIMITED_DEPLETER_ROLE)
+        /// system must be level 1 locked before this function can execute
+        /// asserts system is inside higher level operation when this function is called
+        globalLock(2)
```
Comment (eswak) [inline contracts/limiter/GlobalSystemExitRateLimiter.sol:46]: idk if a global lock is very useful here

File: contracts/limiter/IGSERL.sol [hunk 3]
```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity =0.8.13;
+
+import {IRateLimitedV2} from "../utils/IRateLimitedV2.sol";
+
+interface IGSERL is IRateLimitedV2 {
```
Comment (eswak) [inline contracts/limiter/IGSERL.sol]: I'd prefer full-letters interface names

File: contracts/pcv/utils/ERC20Allocator.sol [hunk 4]
```diff
@@ -55,18 +54,7 @@ contract ERC20Allocator is IERC20Allocator, CoreRefV2, RateLimitedV2 {
...
-    constructor(
-        address _core,
-        uint256 _maxRateLimitPerSecond,
-        uint128 _rateLimitPerSecond,
-        uint128 _bufferCap
-    )
-        CoreRefV2(_core)
-        RateLimitedV2(_maxRateLimitPerSecond, _rateLimitPerSecond, _bufferCap)
-    {}
+    constructor(address _core) CoreRefV2(_core) {}
```
Comment (eswak) [inline contracts/pcv/utils/ERC20Allocator.sol:57]: I really like this new separation of concerns 👍

**Review by eswak — APPROVED** [review]
LGTM, I have some minor comments but I don't see them as blockers to merge

Make sure coverage is 100% :)

File: contracts/limiter/GlobalSystemExitRateLimiter.sol (hunk 2 above)
Comment (ElliotFriedman) [reply contracts/limiter/GlobalSystemExitRateLimiter.sol:46]: there are no external calls in deplete and replenish buffer except to core to validate the sender and their roles. it doesn't cost much to do a warm write to storage, so I don't see any harm in doing it. the only time a user will have to pay this additional gas cost is during a redemption through the non custodial PSM which is a very expensive operation on its own.

**Pull request merged**